src/client.ts

```diff
@@ -1063,4 +1068,50 @@ export class SpvWalletClient {
     }
     return validateTotpForContact(this.xPrivKey, contact, passcode, requesterPaymail, period, digits);
   }
+
+  /**
+   * Syncs merkleroots from the client db to the last known block by SPV-Wallet
+   *
+   * @param {MerkleRootsRepository} repo - Repository interface capable of reading lastEvaluatedKey and saving to the database
+   * @returns void
+   */
+  async SyncMerkleRoots(repo: MerkleRootsRepository, timeoutMs?: number) {
+    const startTime = Date.now();
+
+    let lastEvaluatedKey = await repo.getLastEvaluatedKey();
+    let previousLastEvaluatedKey = lastEvaluatedKey || '';
+    let requestPath = 'merkleroots';
+    let lastEvaluatedKeyQuery = '';
+
+    if (lastEvaluatedKey) {
+      lastEvaluatedKeyQuery = `?lastEvaluatedKey=${lastEvaluatedKey}`;
+    }
+
+    while (true) {
+      if (timeoutMs !== undefined && Date.now() - startTime >= timeoutMs) {
+        this.logger.error('SyncMerkleRoots operation timed out');
+        throw new ErrorSyncMerkleRootsTimeout();
+      }
```

Comment:
After conversation with @dorzepowski & @wregulski we are not so sure if the timeout is necessary. Maybe some kind of lock - to prevent parallel execution of the SYNC...
- or just info in comment description.

Guys @dorzepowski, @wregulski WDYT?

Reply:
For me timeout is good - regarding to our yesterday's conversation I would go for a do while instead of while.
```
      if (merkleRootsResponse.page.lastEvaluatedKey === '') {
        break;
      }

and put this condition into while statement. 
```

Reply:
changed into a do while